server/routerlicious/packages/services-utils/src/redisClientConnectionManager.ts

```diff
@@ -129,4 +163,33 @@ export class RedisClientConnectionManager implements IRedisClientConnectionManag
 		}
 		return this.client;
 	}
+
+	public addErrorHandler(
+		lumberProperties: Record<string, any> = {},
+		errorMessage: string = "Error with Redis",
+		additionalLoggingFunctionality?: (error: Error) => boolean,
+	): void {
+		if (!this.client) {
+			throw new Error("Redis client not initialized");
+		}
+
+		this.client.on("error", (error) => {
+			if (additionalLoggingFunctionality && additionalLoggingFunctionality(error)) {
+				// If the additionalLoggingFunctionality returns true, it means it has completely handled the error
+				return;
+			}
+
+			const commandName: string | undefined =
+				error.commandName ?? error.lastNodeError?.commandName;
+			const args: string[] = error.args ?? error.lastNodeError?.args ?? [];
+			// Grab only the lengths of each arg, to avoid logging sensitive information
+			const argSizes: number[] = args.map((arg) => arg.length);
```

Comment:
I was wondering if we could log the first arg which is typically the key..I think that info will be very helpful to know if it is one command that is failing over and over again or if the impact is more widespread

To be safe, we could only log arg[0] for commands like GET, SET, DEL, HGET, HSET etc so that other like  MULTI, EXEC, or commands that involve complex data structures are not logged

Reply:
also in the set cases, it might be useful to log the size of arg[1]

Reply:
Updated it to log arg[0] raw for GET, SET, DEL, HGET, HSET, and to continue logging the length of the args for every other arg.